test/hotspot/jtreg/compiler/vectorization/TestRoundVectorDoubleRandom.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright (c) 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/**
+ * @test
+ * @key randomness
+ * @bug 8321011
+ * @summary Test vector intrinsic for Math.round(double) in full 64 bits range.
+ *
+ * @library /test/lib /
+ * @run main compiler.vectorization.TestRoundVectorDoubleRandom
+ */
+
+package compiler.vectorization;
+
+import java.util.Random;
+import compiler.lib.ir_framework.DontCompile;
+import compiler.lib.ir_framework.IR;
+import compiler.lib.ir_framework.IRNode;
+import compiler.lib.ir_framework.Run;
+import compiler.lib.ir_framework.RunInfo;
+import compiler.lib.ir_framework.Test;
+import compiler.lib.ir_framework.TestFramework;
+import compiler.lib.ir_framework.Warmup;
+
+public class TestRoundVectorDoubleRandom {
+  private static final int ITERS  = 11000;
+  private static final int ARRLEN = 997;
+  private static final double ADD_INIT = -7500.;
+
+  private static final double[] input = new double[ARRLEN];
+  private static final long [] res = new long[ARRLEN];
+  private static final Random rand = new Random();
+
+  public static void main(String args[]) {
+    TestFramework.runWithFlags("-XX:-TieredCompilation", "-XX:CompileThresholdScaling=0.3", "-XX:MaxVectorSize=16");
+    TestFramework.runWithFlags("-XX:-TieredCompilation", "-XX:CompileThresholdScaling=0.3", "-XX:MaxVectorSize=32");
```

Comment:
Thanks for the suggestion.
Unfortunately, I don't have access to a machine with AVX512, but I do run with a aarch64 via qemu where max vector size > 16, and it works with  "-XX:MaxVectorSize=16".
The reason why the previous test failed (which I fixed in previous commit) with "-XX:MaxVectorSize=8", is because in test framework, it checks the length of vector and make sure it > length of double(8 bytes), i.e. at least 2*(length of Double).

Reply:
Thanks for the suggestion, it's done.
I also tested with "-XX:MaxVectorSize=64" on aarch64 via qemu, it works.